### ISSUE TITLE
in b23 the behavior of queueSpotifyURI: is not defined if the player already has a queued track

### ISSUE THREAD

**Issue by alexcone** [spotify/ios-sdk#774]
## Description
As of b23 the SDK only provides access to and ability to set a single queued URI to be played after the current track completes.  If TrackA is currently playing and one calls queueSpotifyURI: with TrackB and then calls queueSpotifyURI: with TrackC the second call does not seem to simply replace TrackB with TrackC as the next track.  Playback results vary, but sometimes part of TrackB will play before playback switches to TrackC.

### Versions
iOS SDK version b23

### Severity
Major

## Steps to Reproduce
Call SPTAudioStreamingController's playSpotifyURI with TrackA
While the track is playing (after related streamingDelegate calls have happened), call queueSpotifyURI with TrackB
After getting didChangeMetadata shows track has been queued, call queueSpotifyURI with TrackC
Let the player transition from TrackA to next track

### Actual behavior
The streaming controller sometimes plays a part of TrackB before switching to TrackC.  Sometimes it plays all of TrackB and does not switch to TrackC.

### Expected behavior
The second call to queueSpotifyURI: should simply replace the queued track, as there is no replaceQueuedTrack call, nor is there any support (now) for more than one queued track.

**Comment by Voxar** [comment]
Hi @alexcone. The idea is that you can queue several tracks, so the "correct" behaviour should be to finish TrackB before continuing to TrackC. 

We'll look into the discrepancies you mention.

**Comment by alexcone** [comment]
the open question was, given current api, if we need to change the queue so as to go from TrackA to TrackC, there is no way to “un-queue” TrackB.

**Comment by cameronehrlich** [comment]
What is the properway to end the playback of the current song, and enquqe and play the next?

**Comment by elmateo487** [comment]
I also am confused about this. Lets say you queue up 10 songs, and display a list of 10 songs to the user. If the user starts playing at song one, then decides to select song 5 to play it from the list, there is no way to move to song 5. You have to clear the queue and play song 5, at which point you lose the ability to skip forward and backwards in the original queued list.

**Comment by adambolcsfoldi** [comment]
I agree. I don't see how the 'queueSpotifyURI' is useful if there is no way to dequeue a track. Unless you're not letting your user change their mind about what they want to listen to.
